### PR TITLE
Add WildGuard Guardrail Microservice

### PR DESCRIPTION
**Pull request by daniel-de-leon-user293**
## Description

Add WildGuard to guard against privacy, misinformation, harmful language and malicious use on user input prompts and/or output responses generated by LLMs.

## Issues

n/a
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

n/a

## Tests

- Successfully ran:
   - `guardrails_tgi.py` microservice script directly without container
   - with `docker run` CLI
   - with `docker compose`

### CODE REVIEW

**Comment by letonghan** [comment]
Hi @daniel-de-leon-user293 , please add the test script for WildGuard in `test/guardrails` folder like [this](https://github.com/opea-project/GenAIComps/blob/main/tests/guardrails/test_guardrails_llama_guard_langchain_on_intel_hpu.sh). Name the test script as `test_guardrails_wild_guard_langchain_on_intel_hpu.sh` since you use the hpu version of tgi. 
Thanks : )

**Comment by qgao007** [comment]
Hi @daniel-de-leon-user293 , 
Ebi pointed it out to me earlier for my PR, mentioning that the following needs to be added
Also the new Dockerfile introduced by the PR needs to have an entry at:
.github/workflows/docker/compose/guardrails-compose-cd.yaml

File: comps/guardrails/wildguard/langchain/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Guardrails Microservice
+
+To fortify AI initiatives in production, this microservice introduces guardrails designed to encapsulate LLMs, ensuring the enforcement of responsible behavior. With this microservice, you can secure model inputs and outputs, hastening your journey to production and democratizing AI within your organization, building Trustworthy, Safe, and Secure LLM-based Applications.
+
+These guardrails actively prevent the model from interacting with unsafe content, promptly signaling its inability to assist with such requests. With these protective measures in place, you can expedite production timelines and alleviate concerns about unpredictable model responses.
+
+The Guardrails Microservice now offers two primary types of guardrails:
+
+- Input Guardrails: These are applied to user inputs. An input guardrail can either reject the input, halting further processing.
```
[the remark this answers is not in the document]
Comment (daniel-de-leon-user293) [reply comps/guardrails/wildguard/langchain/README.md:9]: It is not needed, thanks for catching!

File: comps/guardrails/wildguard/langchain/README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+```bash
+docker compose -f docker_compose_guardrails.yaml up -d
+```
+
+## 🚀3. Consume Guardrails Service
+
+### 3.1 Check Service Status
+
+```bash
+curl http://localhost:9090/v1/health_check\
```
Comment (ashahba) [inline comps/guardrails/wildguard/langchain/README.md:95]: Please leave an space before `\`:
```
curl http://localhost:9090/v1/health_check \
```

File: comps/guardrails/wildguard/langchain/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  -X GET \
+  -H 'Content-Type: application/json'
+```
+
+### 3.2 Consume Guardrails Service
+
+```bash
+curl http://localhost:9090/v1/guardrails\
```
Comment (ashahba) [inline comps/guardrails/wildguard/langchain/README.md:103]: Same here:
```
curl http://localhost:9090/v1/guardrails \
```

File: tests/guardrails/test_guardrails_wild_guard_langchain_on_intel_hpu.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    result=$(http_proxy= curl http://localhost:5036/v1/guardrails  -X POST   -d '{"text":"How do you buy a car in the US?","parameters":{"max_new_tokens":32}}'   -H 'Content-Type: application/json')
+        if [[ $result == *"car"* ]]; then
+        echo "Result correct."
+    else
+        echo "Result wrong."
+        docker logs test-comps-guardrails-langchain-tgi-server
+        docker logs test-comps-guardrails-langchain-service
+        exit 1
+    fi
+
```
Comment (ashahba) [inline tests/guardrails/test_guardrails_wild_guard_langchain_on_intel_hpu.sh:58]: Remove this empty line

**Review by ashahba — CHANGES_REQUESTED** [review]
Thanks @daniel-de-leon-user293 for this PR!
I only have a few minor change requests.

**Review by ashahba — APPROVED** [review]
LGTM!

**Comment by dcmiddle** [comment]
@daniel-de-leon-user293 Consider adding documentation or commit message explaining why someone would use wildguard instead of llama guard.

@mkbhanda does OPEA have criteria for when it will accept similar features? 
It's nice to have a variety of options if you know what you are looking for but having to choose among options also detracts from ease of use.

**Comment by daniel-de-leon-user293** [comment]
Thank you for your suggestion @dcmiddle. The latest commit adds a bit more description of wild guard in the README. To answer your question,  we’re hoping to provide a diverse set of safety models that users can pick from.

**Comment by dcmiddle** [comment]
> Thank you for your suggestion @dcmiddle. The latest commit adds a bit more description of wild guard in the README. To answer your question, we’re hoping to provide a diverse set of safety models that users can pick from.

cool. so can this be used in conjunction with llama guard? From the description you added it looks like a complementary list of topics.

**Comment by daniel-de-leon-user293** [comment]
No, it can be used in place of Llama Guard. The lists in the README are risk taxonomies that each model was trained to identify according to their respective datasets. Although similar, the models provide different classification performance for difference use cases.

If a user wanted to design an ensemble of guardrails, however, then in that case, these two models could be used in conjunction.

**Comment by sunstonesecure-robert** [comment]
> It's nice to have a variety of options if you know what you are looking for but having to choose among options also detracts from ease of use.

I'd assert it's *essential* to have a variety of options AND you need to know what you are looking for if you are building LLM apps where security+privacy+bias detection > ease of use

**Review by ftian1 — APPROVED** [review]

**Pull request merged**